service/internal/git/gitClient_test.go

```diff
@@ -10,52 +11,71 @@ import (
 	"github.com/google/go-cmp/cmp"
 )
 
-type mockGitSvc struct{}
+type mockGitSvc struct {
+	cloneOpts   *git.CloneOptions
+	fetchOpts   *git.FetchOptions
+	plainOpened bool
+}
 
-func (g mockGitSvc) PlainClone(path string, isBare bool, o *git.CloneOptions) (*git.Repository, error) {
+func (g *mockGitSvc) PlainClone(path string, isBare bool, o *git.CloneOptions) (*git.Repository, error) {
+	g.cloneOpts = o
 	return nil, nil
 }
 
-func (g mockGitSvc) PlainOpen(path string) (*git.Repository, error) {
+func (g *mockGitSvc) PlainOpen(path string) (*git.Repository, error) {
+	g.plainOpened = true
+	if strings.Contains(path, "Err") {
+		return nil, errors.New("plain open goes boom")
```

Comment:
Will do a separate PR to fix test coverage on the git package. This is just ground work, and opening a path to get coverage on the lines I wanted changed.